src/cryptonote_core/cryptonote_format_utils.cpp

```diff
@@ -220,6 +220,26 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------
+  bool generate_key_image_helper_onetime(const account_keys& ack, const crypto::public_key& tx_public_key, const crypto::secret_key* onetime_c, size_t real_output_index, keypair& in_ephemeral, crypto::key_image& ki, keypair& in_ephemeral2, crypto::key_image& ki2)
+  {
+    if (!generate_key_image_helper(ack, tx_public_key, real_output_index, in_ephemeral, ki))
+      return false;
+    if (onetime_c)
+    {
+      crypto::secret_key d;
+      crypto::public_key D;
+      crypto::hash_to_scalar(onetime_c->data, HASH_SIZE, reinterpret_cast<crypto::ec_scalar&>(d));
+      secret_key_to_public_key(d, D);
+      account_keys ack2;
+      ack2.m_view_secret_key = *onetime_c;
+      sc_add((unsigned char*)&ack2.m_spend_secret_key, (unsigned char*)&ack.m_spend_secret_key, (unsigned char*)&d);
```

Comment:
Cast `(unsigned char*)` here is really necessary. If I use `key.data`, the compiler emits an error: `no known conversion from 'char [32]' to 'unsigned char *'`